src/publish.proj

```diff
@@ -23,16 +23,17 @@
   <PropertyGroup>
     <PublishPattern Condition="'$(PublishPattern)' == '' and '$(PublishTestNativeBins)' != 'true'">$(PackagesBinDir)**\*.nupkg</PublishPattern>
     <PublishPattern Condition="'$(PublishPattern)' == '' and '$(PublishTestNativeBins)' == 'true'">$(OutputPath)\tests\src\**</PublishPattern>
+    <DistroRid Condition="'$(__DistroRid)' != ''">$(__DistroRid.Replace(".","-"))</DistroRid>
```

Comment:
Have you checked whether $(DistroRid) is used anywhere else?    Since a Rid is used for all sorts of things (like constructing paths) I'd prefer you did this replace() on the final ContainerName and not the variables used to construct it, as that can't possibly change other behaviors downstream of changing the value of a property.